Extension/src/LanguageServer/extension.ts

```diff
@@ -1413,13 +1413,33 @@ export async function preReleaseCheck(): Promise<void> {
     }
 }
 
-export async function getIncludes(maxDepth: number): Promise<any> {
-    const tokenSource = new vscode.CancellationTokenSource();
-    const includes = await clients.ActiveClient.getIncludes(maxDepth, tokenSource.token);
-    tokenSource.dispose();
+export async function getIncludesWithCancellation(maxDepth: number, token: vscode.CancellationToken): Promise<GetIncludesResult> {
+    const includes = await clients.ActiveClient.getIncludes(maxDepth, token);
+    const wksFolder = clients.ActiveClient.RootUri
```

Comment:
I don't believe it's possible for the RootUri to not refer to the primary workspace folder.  Maybe this filtering isn't needed?

Reply:
Thanks! I dropped the search for the current workspace, and assume that the RootUri is the one.